sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -1741,9 +1740,8 @@ public void populateDisplayData(DisplayData.Builder builder) {
     }
 
     @Override
-    public PDone expand(PCollection<T> input) {
-      input.apply(inner);
-      return PDone.in(input.getPipeline());
+    public WriteFilesResult<?> expand(PCollection<T> input) {
```

Comment:
This solution looks ok, but I am wary of the consequences of changing the return type. It looks like we introduced TypedWrite on Beam for the achieve the same goal without breaking backwards compatibility.

The 'modern' preferred way to write files on Beam is via `FileIO.write()` which already returns `WriteFilesResult`.
Can you check if you we can achieve the intended results by relying on FileIO.write() + AvroIO.sink() ? or is there anything missing?

CC: @jkff in case you have some extra detail to add.

Reply:
Agreed, this should use FileIO.write + AvroIO.sink - the current change is incompatible and will break anybody's transforms of the form:

```
PDone expand(...) {
  ...
  return AvroIO.write()...;  // If return type changes to WFR, this stops compiling
}
```